test/python/circuit/test_initializer.py

```diff
@@ -367,6 +367,44 @@ def test_from_int(self):
         actual_sv = Statevector.from_instruction(qc)
         self.assertTrue(desired_sv == actual_sv)
 
+    def _remove_resets(self, circ):
+        circ.data = [tup for tup in circ.data if tup[0].name != 'reset']
+
+    def test_global_phase_random(self):
```

Comment:
Isn't the other test enough? Or does the algorithm of the intialize-instruction work differently for more than 1 qubit and therefore we should also test that? If not I think the other test already covers all cases we've seen in the past 🙂

Reply:
This tests when the angle lists have >= 1 element with irrational values.

Reply:
Hi @Cryoris and @ewinston! Could the reset from the initialize be removed?
If yes, initialize can be converted to a gate allowing, for instance, controlled initialization used in some algorithms.

Reply:
@adjs This was discussed in [issue #1896](https://github.com/Qiskit/qiskit-terra/issues/1896) but it doesn't seem needed here. Perhaps it could be added as an optional boolean parameter although someone may question whether it changes the meaning of initialization. One other option for that case it to add something like a `remove_instructions` method  to QuantumCircuit.

Reply:
We discussed this a few times in the team and if I recall correctly the consensus was to not remove them, since the initialization should guarantee that after it is executed the statevector is in the desired state. If there are preceding gates in the circuit, and the resets are accidentally forgotten, we can't guarantee the statevector. 

If you want to remove the resets you can use the `RemoveResetInZeroState` transpiler pass:
```python
>>> from qiskit import QuantumCircuit
>>> from qiskit.transpiler.passes import RemoveResetInZeroState
>>> circuit = QuantumCircuit(2)
>>> circuit.initialize([1, 0, 0, 0], [0, 1])
>>> without_resets = RemoveResetInZeroState()(circuit.decompose()).draw()
```

Reply:
Hi @Cryoris. Thank you for the answer, and I understood that this was already discussed and this PR should go without removing the resets. I raised this question because an alternative to go from a state | a> to a state | b> is to apply the inverse of initialize |a> followed by initialize |b>. Another example of initialization without resets is a controlled initialization as in arXiv:1703.10793. In these cases a gate initialize should be more useful than an instruction with the resets.

Reply:
Thanks for the explanations @adjs, could you either re-open #1896 or open a new issue where you describe your points? I think we could also have a `StatePreparation` class that has no resets and a separate `Initialize` that uses resets + state-preparation.